eo-maven-plugin/src/main/java/org/eolang/maven/MjParse.java

```diff
@@ -146,27 +146,47 @@ private Node parsed(final Path source, final String identifier) throws IOExcepti
             "Parsed program '%s' from %[file]s:\n %s",
             identifier, this.sourcesDir.toPath().relativize(source.toAbsolutePath()), xmir
         );
-        final String name = new ObjectName(xmir).get();
+        String name = "";
         final Node document = xmir.inner();
+        try {
+            name = new ObjectName(xmir).get();
+        } catch (final IllegalStateException exception) {
+            MjParse.applyError("mandatory-object-name", exception.getMessage(), document);
+        }
         if (!name.equals(identifier)) {
-            new Xembler(
-                new Directives()
-                    .xpath("/object")
-                    .addIf("errors")
-                    .add("error")
-                    .attr("check", "validate-object-name")
-                    .attr("severity", "critical")
-                    .set(
-                        String.format(
-                            "Tojo identifier '%s' does not match to result object name '%s'",
-                            identifier, name
-                        )
-                    )
-            ).applyQuietly(document);
+            MjParse.applyError(
+                "validate-object-name",
+                String.format(
+                    "Tojo identifier '%s' does not match to result object name '%s'", identifier,
+                    name
+                ),
+                document
+            );
         }
         return document;
     }
 
+    /**
+     * Apply error to the document.
+     * @param check Check name
+     * @param message Error message
+     * @param document Document
+     * @return Document with applied errors
+     */
+    private static Node applyError(
```

Comment:
@h1alexbel read this: https://www.yegor256.com/2018/08/22/builders-and-manipulators.html